### PR TITLE
[cpp rest-sdk] Reponse headers handler

### PR DESCRIPTION
**Pull request by whoan**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.4.x`, `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Support processing the HTTP response headers through a handler that can be set in the `ApiClient` class.

Some APIs ([like this](https://testnet.bitmex.com/app/restAPI#Viewing-Your-Request-Rate-Limit)) return useful information in the response headers, so this PR is to support processing them.

Ping @ravinikam @stkrwork @fvarose @etherealjoy @martindelille

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/cpp-rest-sdk-client/apiclient-header.mustache [hunk 1]
```diff
@@ -32,6 +33,9 @@ public:
     ApiClient( std::shared_ptr<ApiConfiguration> configuration = nullptr );
     virtual ~ApiClient();
 
+    const ResponseHeadersHandlerType& getResponseHeadersHandler() const;
```
Comment (etherealjoy) [inline modules/openapi-generator/src/main/resources/cpp-rest-sdk-client/apiclient-header.mustache]: I cannot find this typedef, where is it?

**Comment by etherealjoy** [comment]
What about error status? no need for headers there?

File: modules/openapi-generator/src/main/resources/cpp-rest-sdk-client/apiclient-header.mustache (hunk 1 above)
Comment (whoan) [reply modules/openapi-generator/src/main/resources/cpp-rest-sdk-client/apiclient-header.mustache]: @etherealjoy your are right, I forgot that. Now it is fixed.

**Comment by whoan** [comment]
@etherealjoy If I understand your question(s), with the status code is not enough to get all the information some APIs send. As an example, the one I am using send information about remaining requests in the headers.

**Comment by etherealjoy** [comment]
@whoan 
Did you compile the sample code?

**Comment by whoan** [comment]
@ethanjcohen Yes, I compiled the code. Before the last commit I hadn't done, but now I did and it is working fine.

Regarding the status code, changing the handler signature to receive it would be a good improvement indeed. Do you want me to do it in this PR?

**Comment by etherealjoy** [comment]
@whoan 
You can do it later.

**Comment by wing328** [comment]
@whoan thanks for the PR, which has been included in the v3.3.4 release: https://twitter.com/oas_generator/status/1068772409795207168

**Pull request merged**